### PR TITLE
Activity RichSubject

### PR DESCRIPTION
**Pull request by aleister09**
This PR implement RichSubject to fix the issue 
[#971](https://github.com/nextcloud/android/issues/971)

### CODE REVIEW

**Comment by mario** [comment]
Afaik rich objects are part of Activities v2 api which is related to the activities app. In case where we're still on v1 activities, it just won't have rich objects but I do agree we need to query capabilities to see what we have. 

As for OCS v2, it was (I think) available in NC9 so I think we're safe there.

File: src/com/owncloud/android/lib/resources/activities/models/RichSubject.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.ArrayList;
+
+/**
+ * RichSubject Data Model
+ */
+
+public class RichSubject {
+
+    public String richSubject;
+    ArrayList<RichObject> richObjectList=new ArrayList<>();
```
Comment (mario) [inline src/com/owncloud/android/lib/resources/activities/models/RichSubject.java]: Please make this List<RichObject> richObjecstList and some separation between name and = sign.

**Comment by mario** [comment]
Commented a bit. Also additional comment:

Please rename RichSubject to RichElement or something because this also can be used for message_rich. Would appreciate if you add support for that as well.

**Comment by AndyScherzinger** [comment]
👍 fine by me

**Pull request merged**